pkg/chunked/filesystem_linux.go

```diff
@@ -49,10 +49,33 @@ type fileMetadata struct {
 	skipSetAttrs bool
 }
 
+// splitPath takes a file path as input and returns two components: dir and base.
+// Differently than filepath.Split(), this function handles some edge cases.
+// If the path refers to a file in the root directory, the returned dir is the
+// empty string.
+// The returned base value is never empty, it never contains any slash and the
+// value "..".
+func splitPath(path string) (string, string, error) {
+	path = filepath.Clean(path)
+	dir, base := filepath.Split(path)
+	dir = cleanAbsDirectory(dir)
+	if base == "" || base == ".." {
+		base = "."
+	}
```

Comment:
```suggestion
	path = cleanAbsDirectory(path)
	dir, base := filepath.Split(path)
	if base == "" {
		base = "."
	}
```
would do the same thing a bit simpler way.

Reply:
the difference is that it won't remove trailing slashes from `dir`, which probably doesn't matter.  I'll adjust to your version

Reply:
Actually, wouldn’t a trailing slash break the `openOrCreateDirUnderRoot` calls to create a parent directory? `filepath.Dir("/a/")) == "/a"`. So either keep the code as is, or call `cleanAbsDirectory` twice, or…

[Somewhere in here is a discussion about using `filepath.Dir`/`filepath.Base` instead of the fairly naive `filepath.Split`, but I suspect neither of us has much taste for another complete rewrite.]